### PR TITLE
feat(si): add specific power quantity

### PR DESCRIPTION
**Pull request by swaits**
Specific power is used to represent a power-to-weight ratio, like in vehicles or human power; as in watts per kilogram or ft-lbf/s per pound.

Added the `SpecificPower` quantity using the existing `Power` quantity as a guideline.

For more information on this quantity, please see a description of [specific power](https://en.wikipedia.org/wiki/Power-to-weight_ratio) on Wikipedia.

### CODE REVIEW

File: src/si/specific_power.rs [hunk 1]
```diff
@@ -0,0 +1,112 @@
+//! Specific Power (base unit watt per kilogram or m² · s⁻³).
+
+quantity! {
+    /// Specific Power (base unit watt per kilogram or m² · s⁻³).
+    quantity: SpecificPower; "specific power";
+    /// Dimension of specific power, L²T⁻³ (base unit watt per kilogram or m² · s⁻³).
```
Comment (iliekturtles) [inline src/si/specific_power.rs:6]:
```suggestion
//! Specific power (base unit watt per kilogram, m² · s⁻³).

quantity! {
    /// Specific power (base unit watt per kilogram, m² · s⁻³).
    quantity: SpecificPower; "specific power";
    /// Dimension of specific power, L²T⁻³ (base unit watt per kilogram, m² · s⁻³).
```

**Review by iliekturtles — COMMENTED** [review]
Thanks for the PR! I still need to do a more in depth review, and have started test actions in the meantime.

Some of the units have multiple divisions (e.g. erg per second per kilogram). Are these commonly used this way? I've only skimmed the referenced article at this point, but generally multiple divisions avoided: https://www.nist.gov/pml/special-publication-811/nist-guide-si-chapter-6-rules-and-style-conventions-printing-and-using#616

**Comment by swaits** [comment]
I really only know of specific power's use in `W/kg`, `(lbf/<unit time>)/lb`, and `hp/lb`. I'm completely aligned with deleting things like `(erg/s)/kg` as I only included it because I started with `power.rs` as my baseline (i.e. it's copypasta). If we're wrong and someone needs it in the future, they can ask for it or submit their own PR.

**Comment by swaits** [comment]
Oh... that said, I felt it important to include this one...

```rust
// Less commonly used, but more accurate (reduced) derived unit of specific power.
@square_meter_per_cubic_second: 1.0_E0; "m² · s⁻³", "square meter per cubic second", "square meters per cubic second";
```

... because that's the true base unit, even though it's not as commonly used as multiplied by `kg/kg` to look like `W/kg`.

**Comment by swaits** [comment]
Please LMK if there's anything else I can do to help make this one acceptable for merge. And thank you for considering it!

**Comment by iliekturtles** [comment]
I reviewed and pushed a commit to your fork. Please review, rebase, and squash the commits!
                                                            
 * Formatting for readability.
 * Fix conversion factor used for pounds.
 * Check all units based on the units they are derived from.
   * Add `check_units_energy` test.

**Comment by swaits** [comment]
Hey, yes, I'll try to get to this today... I'm pretty buried in `$DAYJOB` but excited to figure this one out.

**Pull request closed without merging**